lib/internal/modules/esm/loader.js

```diff
@@ -429,7 +429,23 @@ function createModuleLoader(useCustomLoadersIfPresent = true) {
   return new DefaultModuleLoader();
 }
 
+/**
+ * Get the HooksProxy instance. If is not defined creates a new one.
+ * @returns {HooksProxy}
+ */
+function getHooksProxy() {
```

Comment:
It looks like this duplicates logic from the `CustomizedModuleLoader` constructor in this file. Could you replace that logic with a call to this function.